metrics/web/test/common/presentation/widgets/build_status_view_test.dart

```diff
@@ -0,0 +1,279 @@
+
+        await mockNetworkImagesFor(() {
+          return tester.pumpWidget(
+            _BuildStatusViewTestbed(
+              strategy: strategy,
+              buildStatus: expectedBuildStatus,
+            ),
+          );
+        });
+
+        final valueNetworkImage = tester.widget<ValueNetworkImage>(
+          valueNetworkImageFinder,
+        );
+
+        expect(valueNetworkImage.value, equals(expectedBuildStatus));
+      },
+    );
+
+    testWidgets(
+      "applies the given height to the value network image",
+      (WidgetTester tester) async {
+        const expectedHeight = 24.0;
+        when(strategy.getAsset(finishedBuildStatus)).thenReturn(asset);
+
+        await mockNetworkImagesFor(() {
+          return tester.pumpWidget(
+            _BuildStatusViewTestbed(
+              strategy: strategy,
+              buildStatus: finishedBuildStatus,
+              height: expectedHeight,
+            ),
+          );
+        });
+
+        final valueNetworkImage = tester.widget<ValueNetworkImage>(
+          valueNetworkImageFinder,
+        );
+
+        expect(valueNetworkImage.height, equals(expectedHeight));
+      },
+    );
+
+    testWidgets(
+      "applies the given width to the value network image",
+      (WidgetTester tester) async {
+        const expectedWidth = 24.0;
+        when(strategy.getAsset(finishedBuildStatus)).thenReturn(asset);
+
+        await mockNetworkImagesFor(() {
+          return tester.pumpWidget(
+            _BuildStatusViewTestbed(
+              strategy: strategy,
+              buildStatus: finishedBuildStatus,
+              width: expectedWidth,
+            ),
+          );
+        });
+
+        final valueNetworkImage = tester.widget<ValueNetworkImage>(
+          valueNetworkImageFinder,
+        );
+
+        expect(valueNetworkImage.width, equals(expectedWidth));
+      },
+    );
+  });
+}
+
+/// A testbed class that is used to test the [BuildStatusView] widget.
+class _BuildStatusViewTestbed extends StatelessWidget {
+  /// A [BuildStatus] to use in tests.
+  final BuildStatus buildStatus;
+
+  /// A [ValueBasedAssetStrategy] to use in tests.
+  final ValueBasedAssetStrategy<BuildStatus> strategy;
+
+  /// A height to use in tests.
+  final double height;
+
+  /// A width to use in tests.
+  final double width;
+
+  /// Creates a new instance of the [_BuildStatusViewTestbed] with the
+  /// given parameters.
+  const _BuildStatusViewTestbed({
+    Key key,
+    this.strategy,
+    this.buildStatus,
+    this.height,
+    this.width,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      home: RiveAnimationTestbed(
+        child: BuildStatusView(
+          strategy: strategy,
+          buildStatus: buildStatus,
+          height: height,
+          width: width,
+        ),
+      ),
+    );
+  }
+}
+
+class _ValueBasedAssetStrategyMock<T> extends Mock
+    implements ValueBasedAssetStrategy<T> {}
```

Comment:
As this is a private mock and we sure we need it for build statuses, can we reduce generics here?
```suggestion
class _ValueBasedAssetStrategyMock extends Mock
    implements ValueBasedAssetStrategy<BuildStatus> {}
```